packages/nimble-components/src/utilities/testing/component.ts

```diff
@@ -46,44 +46,22 @@ export async function sendKeyDownEvents(
 /** A helper function to abstract turning waiting for an event to fire into a promise.
  * The returned promise should be resolved prior to completing a test.
  */
-export async function waitForEventAsync(
+export async function waitForEvent<T extends Event>(
     element: HTMLElement,
-    eventName: string
+    eventName: string,
+    callback?: (evt: T) => void
 ): Promise<void> {
     return new Promise(resolve => {
-        const handler = (): void => {
-            element.removeEventListener(eventName, handler);
+        const handler = ((evt: T): void => {
+            callback?.(evt);
             resolve();
-        };
-        element.addEventListener(eventName, handler);
+        }) as EventListener;
```

Comment:
Here we can encapsulate the lie to addEventListener about the typing on the callback so waitForEvent is a little cleaner than using addEventListener directly.